source/backend/api/Services/DocumentFileService.cs

```diff
@@ -243,10 +243,10 @@ public async Task<ExternalResponse<string>> DeleteResearchDocumentAsync(PimsRese
 
             DeleteQueuedDocumentItem(currentDocument.DocumentId);
             _researchFileDocumentRepository.DeleteResearch(researchFileDocument);
-            DeleteDocument(currentDocument);
-
             _documentRepository.SaveChanges();
-            transaction.Commit();
+
+            DeleteDocument(currentDocument);
```

Comment:
it seems like the on-site sql server was choosing to execute the delete on the document table before the delete on the document_file table, attempting to move this statement after saveChanges to resolve that.

Reply:
I added a similar function for Management File documents. Could you please update/rebase your branch and apply the fix to Management File documents as well? Thanks